manual/manual/cmds/instrumented-runtime.etex

```diff
@@ -0,0 +1,148 @@
+\chapter{Runtime tracing with the instrumented runtime}
+%HEVEA\cutname{instrumented-runtime.html}
+
+This chapter describes the OCaml instrumented runtime, a runtime variant
+allowing the collection of events and metrics.
+It aims to provide insight into a program's execution.
+
+\section{s:instr-runtime-overview}{Overview}
+
+Once compiled and linked with the instrumented runtime, any OCaml program
+can generate {\em trace files} that can then be read
+and analyzed by users in order to understand specific runtime behaviors.
+
+{\em Common Trace Format} is the general purpose binary
+format used to encode traces.
+A complete trace is comprised of:
+\begin{itemize}
+\item a {\em metadata file}, part of the OCaml distribution
+\item and a {\em trace file}, generated by the runtime\
+  in the program being traced.
+\end{itemize}
+
+For more information on {\em Common Trace Format}, see the website at
+\ifouthtml
+\ahref{https://diamon.org/ctf/}{https://diamon.org/ctf/}.
+\else
+{\tt https://diamon.org/ctf/}
+\fi
+
+\section{s:instr-runtime-enabling}{Enabling runtime instrumentation}
+
+Compiling a program and linking it with the instrumented runtime,
+can be done with {\tt -runtime-variant} flag {\tt i}:
+
+\begin{verbatim}
+       ocamlopt -runtime-variant i program.ml -o program
+\end{verbatim}
+
+The resulting program can then be traced by running it with the environment
+variable {\tt OCAML\_EVENTLOG\_ENABLED}:
+
+\begin{verbatim}
+        OCAML_EVENTLOG_ENABLED=1 ./program
+\end{verbatim}
+
+The program will execute and generate a trace file in the
+current directory.
+
+\subsection{ss:instr-runtime-prefix}{Trace filename}
+
+The default trace filename is {\tt caml-\{pid\}-eventlog}, where {\tt \{pid\}}
+is the process identifier of the traced program.
+
+This filename can also be specified using the
+{\tt OCAML\_EVENTLOG\_FILE} environment variable.
+The given path will be suffixed with {\tt .pid.eventlog}.
+
+\begin{verbatim}
+        OCAML_EVENTLOG_FILE=/tmp/some/prefix OCAML_EVENTLOG_ENABLED=1 ./program
```

Comment:
I agree that the name could be changed to `_PREFIX`, however my approach on allowing the user to explicitly set a full filename is a conservative one, considering in future designs (when executing on OCaml Multicore, with multiple domains), there likely won't be one, but many traces generated.
With this usecase in mind, I would rather avoid implementing a feature that may be obsoleted in the future. I considered the "pid" to provide a good enough indicator for scripting purpose.

Reply:
I'm okay with `_PREFIX` and no way to set a specific file. Maybe it could be useful to document the reason for this scheme, so that in particular script authors know that they should look for `$OCAML_EVENTLOG_FILE_PREFIX.*.eventlog`, and not assume a single file, if they want to be robust to future changes.

Reply:
Sure, how should we proceed with the _PREFIX change? Should I open another PR to trunk?
I will document as well the reason for this change.
I will as well change the default naming scheme to `caml-{pid}.eventlog` to make everything more consistent.